docs/event-processing/idempotent-reader.md

```diff
@@ -0,0 +1,52 @@
+
+2. *Incorrect Application Logic*: An [Event Source](../event-source/event-source.md) could mistakenly produce the same event multiple times, populating the [Event Stream](../event-stream/event-stream.md) with multiple distinct Events from the perspective of the [Event Streaming Platform](../event-stream/event-streaming-platform.md). For example, imagine a bug in the Event Source that writes a customer payment Event twice, instead of just once. The Event Streaming Platform knows nothing of the business logic, so it cannot differentiate between the two Events and instead considers them as two distinct payment Events.
+
+## Problem
+How can an application deal with duplicate Events when reading from an Event Stream?
+
+## Solution
+![idempotent-reader](../img/idempotent-reader.svg)
+
+This can be addressed using exactly-once semantics (EOS), including native support for transactions and support for idempotent clients.
+EOS allows [Event Streaming Applications](../event-processing/event-processing-application.md) to process data without loss or duplication, ensuring that computed results are always accurate. 
+
+[Idempotent Writing](idempotent-writer.md) by the [Event Source](../event-source/event-source.md) is the first step in solving this problem. Idempotent Writing provides strong, exactly-once delivery guarantees of the producer's Events, and removes operational failures as a cause of duplicate Events.
+
+On the reading side, in [Event Processors](../event-processing/event-processor.md) and [Event Sinks](../event-sink/event-sink.md), an Idempotent Reader can be configured to read only committed transactions. This prevents Events within incomplete transactions from being read, providing the reader isolation from operational writer failures. Keep in mind that idempotency means that the reader's business logic must be able to process the same consumed Event multiple times, without generating side-effects or otherwise incorrectly updating its internal state. 
+
+Duplicates caused by incorrect application logic are best resolved by fixing the application's logic. In cases where this is not possible, such as when event generation is triggered by external processes, the next best option is to tag duplicate events with tracking IDs. A tracking ID should be a field that is unique to the logical event, such as an event key or request ID. The consumer can then read the tracking ID, cross-reference it against an internal state store of IDs it has already processed, and discard the event if necessary.
+
+
+## Implementation
```

Comment:
Unlike the Idempotent Writer pattern, this pattern doesn't say a word about the plain consumer clients (= no ksqlDB, no KStreams).

Reply:
@miguno please see latest commit